File: parent-project/spring_boot_microservice_course_creation/src/main/java/com/example/spring_boot_microservice1_course/service/CourseService.java
package com.example.spring_boot_microservice1_course.service;

import com.example.spring_boot_microservice1_course.model.Course;

import java.util.List;

public interface CourseService {
    Course saveCourse(Course course);

    void deleteCourse(Long courseId);

    List<Course> findAllCourse();
}

File: react-course-seller-master/src/services/course.service.js
import { BASE_API_URL } from '../common/Constants';
import axios from 'axios';
import { authHeader } from './base.service';


const API_URL = BASE_API_URL + '/gateway/course';
const API_URL_CREATE_COURSE =API_URL+ '/createcourse';
const API_URL_DELETE_COURSE =API_URL+ '/deletecourse';
const API_URL_GET_COURSE =API_URL+ '/getcourses';
class CourseService {

    saveCourse(course) {
        return axios.post(API_URL_CREATE_COURSE, course, {headers: authHeader()});
    }

    deleteCourse(course) {
        return axios.delete(API_URL_DELETE_COURSE + '/' + course.id, {headers: authHeader()});
    }

    getAllCourses() {
        return axios.get(API_URL_GET_COURSE);
    }

}

export default new CourseService();
